stockfish/models.py

```diff
@@ -552,10 +557,10 @@ def get_top_moves(self, num_top_moves: int = 5) -> List[dict]:
             elif (
                 ("multipv" in current_line)
                 and ("depth" in current_line)
-                and current_line[current_line.index("depth") + 1] == self.depth
+                and ("nodes" in current_line)
             ):
-                multiPV_number = int(current_line[current_line.index("multipv") + 1])
-                if multiPV_number <= num_top_moves:
+                multipv_num = int(current_line[current_line.index("multipv") + 1])
+                if multipv_num <= num_top_moves and len(top_moves) < num_top_moves:
```

Comment:
If we validate whether we need to examine more moves via the number of moves already captured (which I personally like a lot :+1:), then it would still make sense to include an `else: break` for this if-statement as well, so that the loop doesn't go through all the remaining lines unnecessarily. Or is there some case against this that I don't see?

Reply:
I don't remember the details here, but since there are two conditions to evaluate, it does not necessarily follow that the search should stop if one of them is true? Let's say `len(top_moves) < num_top_moves` is `False`, we still want to continue search until both `multipv_num <= num_top_moves and len(top_moves) < num_top_moves` are `True`?

I haven't looked at the raw Stockfish output for a year, so I could be confused here. 😬

Reply:
Would also note that this is hardly a performance issue, as this is just a loop reading pure text, and there are only a handful of lines to read — so 1-5 ms? Although, of course — in principle — if it's superfluous, it should not run. But I'd have to see it in action to make an evaluation, and would err on the side of caution.

Reply:
I'm actually not quite sure why the first part of the if-statement is there. In my opinion, if everything works, it should always evaluate to `True`. But anyway, even if I'm wrong there, it would definitely make sense to introduce an `elif len(top_moves) >= num_top_moves: break` so we don't spend unnecessary time in the loop

Reply:
> Would also note that this is hardly a performance issue, as this is just a loop reading pure text, and there are only a handful of lines to read — so 1-5 ms? Although, of course — in principle — if it's superfluous, it should not run. But I'd have to see it in action to make an evaluation, and would err on the side of caution.

Sure, I'm probably being a little petty there 😄 . But if we could avoid it, I would still like to avoid it

Reply:
> I'm actually not quite sure why the first part of the if-statement is there. In my opinion, if everything works, it should always evaluate to `True`. 

I'm afraid I don't understand. `multipv_num` is which principle variation the engine is evaluating, isn't it? So it could be the 20th, while we only want the top 10 lines, ie the first condition would be `False`? I think that's a necessary condition. (Although, I could be wrong. I don't have raw Stockfish output to look at, so I'm relying heavily on memory from last year...)

> But anyway, even if I'm wrong there, it would definitely make sense to introduce an `elif len(top_moves) >= num_top_moves: break` so we don't spend unnecessary time in the loop

Yes, adding an `elif` makes sense, and should do no harm. Will add 👍

Reply:
Ah, this is what you are referring to, I guess 😄. 

```py
self._parameters.update({"MultiPV": num_top_moves})
```

Reply:
> Yes, adding an `elif` makes sense, and should do no harm. Will add 👍

Yes, I think this is a very simple and safe option

> `multipv_num` is which principle variation the engine is evaluating, isn't it? So it could be the 20th, while we only want the top 10 lines, ie the first condition would be `False`? I think that's a necessary condition.

Just for completeness (it's not that relevant now anyway if we use an elif), but to explain what I meant: You're absolutely right. `multipv_num` indicates the line that Stockfish is evaluating. But the point is that the wrapper sets the MultiPV value [at the start of the function](https://github.com/py-stockfish/stockfish/blob/7a10c80a1133e5f092ec98c8f33b0451c9ad0121/stockfish/models.py#L540)  to the number of requested moves to get enough lines. But with that, according to my understanding (and according to my tests), the first part of the if-query should always be true.

Reply:
> Ah, this is what you are referring to, I guess 😄.
> 
> ```python
> self._parameters.update({"MultiPV": num_top_moves})
> ```

Yep 😄